src/joystick/hidapi/SDL_hidapi_8bitdo.c

```diff
@@ -236,7 +256,7 @@ static void HIDAPI_Driver8BitDo_HandleStatePacket(SDL_Joystick *joystick, SDL_Dr
 {
     Sint16 axis;
     Uint64 timestamp = SDL_GetTicksNS();
-    if (data[0] != 0x03 && data[0] != 0x01) {
+    if (data[0] != 0x03 && data[0] != 0x01 && data[0] != 0x04) {
```

Comment:
Does the first byte vary depending on controller type?  Maybe we should add the report ID as part of the SDL_Driver8BitDo_Context structure and validate that the packet matches the expected value for the current controller?

Reply:
It is not the type of controller, but the 04 report created for SN30 PRO/SF30 PRO/PRO 2 compatibility with previous versions of the report.
In the future, products that are compatible with SDL will be adapted using the 04 report.